### DISCUSSION TITLE
Helm Chart for Photoprism

### DISCUSSION THREAD

**Discussion by veritasecops** [photoprism/photoprism#3460] [Q&A]
I would like to use Helm and k3s for photoprism. I found a Helm chart in github but has anyone used it or done this yet?

**Comment by virot** [comment]
There seems to be helmcharts for Photoprism now.. But they are also kinda broken if you have a nitpicking K8s.

2025-12-28T17:32:11.368890112Z: warning: Upgrade "photoprism" failed: failed to create resource: PersistentVolumeClaim "photoprism-photoprism-storage" is invalid: metadata.labels: Invalid value: "photoprism-plus-1.0.1+251130-b3068414c": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')

**Reply by lastzero** [reply]
They are not broken when used with Rancher/k3s. It may be that other tools have issues with the version string we are using. To resolve this issue, please suggest a valid version string that contains the same information. Also, let us know what software reports the error. Thank you very much!

**Reply by lastzero** [reply]
@veritasecops see our docs at https://docs.photoprism.app/getting-started/advanced/kubernetes/

**Reply by virot** [reply]
> Also, let us know what software reports the error.

I am running Talos cluster with the latest helm and k8s version. 

Talos is annoyingly strict, but have a + sign in a label version is against the k8s concept. 

[Labels and Selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)

> Valid label value:
> - must be 63 characters or less (can be empty),
> - unless empty, must begin and end with an alphanumeric character ([a-z0-9A-Z]),
> - could contain dashes (-), underscores (_), dots (.), and alphanumerics between.

I'll update the version in my HelmRelease to follow standard. Do you have a repo for the charts, so we could easier look and create merge requests. Or should we have discussions in the main repo (here). This issue is to be closed in my world, as the issue was lack of helm chart which is not lacking.

**Reply by virot** [reply]
I tried on my old K3s cluster.
> Helm upgrade failed for release photoprism/photoprism with chart photoprism-plus@0.1.1+251130-b3068414c: failed to create resource: PersistentVolumeClaim "photoprism-photoprism-storage" is invalid: metadata.labels: Invalid value: "photoprism-plus-0.1.1+251130-b3068414c": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')

root@cluster01:/# /usr/local/bin/k3s --version
k3s version v1.32.2+k3s1 (381620ef)
go version go1.23.6
root@cluster01:/# helm version --short
v3.19.4+g7cfb6e4

**Reply by virot** [reply]
So. Looking at: [Helm General Conventions](https://helm.sh/docs/chart_best_practices/conventions/)
> When SemVer versions are stored in Kubernetes labels, we conventionally alter the + character to an _ character, as labels do not allow the + sign as a value.

I suggest following their recommendation in _helpers.tpl, we update photoprism-plus.chart to the below one. Minimum change, but will work with newest helm and k3s/k8s.

```
{{- define "photoprism-plus.chart" -}}
{{- printf "%s-%s" .Chart.Name (replace .Chart.Version "+" "_") -}}
{{- end -}}
```

**Reply by lastzero** [reply]
In the newly uploaded chart version, the "+" was replaced with a "-". We previously used a plus sign because we saw it used in official SuSE/Rancher charts, and because it's a common way to include upstream/image versions in SemVer version strings.

Please let us know if that works for you. If not, it's possible to drop the image version entirely.

Should you or others want to contribute to the charts, we can move them to our public repo in a next step. Have a good start into the new year! 🚀✨
